addons/mail/static/src/core/messaging_service.js

```diff
@@ -191,20 +191,29 @@ export class Messaging {
                     break;
                 case "discuss.channel/leave":
                     {
-                        const thread = this.threadService.insert({
-                            ...notif.payload,
-                            model: "discuss.channel",
-                        });
-                        this.threadService.remove(thread);
-                        if (thread.localId === this.store.discuss.threadLocalId) {
-                            this.store.discuss.threadLocalId = undefined;
-                        }
+                        const thread = this._removeThread(notif.payload);
                         this.notificationService.add(
                             sprintf(_t("You unsubscribed from %s."), thread.displayName),
                             { type: "info" }
                         );
                     }
                     break;
+                case "discuss.channel/delete":
+                    {
+                        const thread = this._removeThread({ id: notif.payload.id });
+                        this.store.discuss.starred.messages = this.store.discuss.starred.messages.filter(
+                            (message) => message.resId !== thread.id
+                        );
+                        this.store.discuss.starred.counter = this.store.discuss.starred.messages.length;
+                        this.store.discuss.inbox.messages = this.store.discuss.inbox.messages.filter(
+                            (message) => message.resId !== thread.id
+                        );
+                        this.store.discuss.inbox.counter = this.store.discuss.inbox.messages.length;
+                        thread.messages.forEach((message) => {
```

Comment:
Recomputing the counter in starred and inbox based on amount of fetched messages is wrong.
If the user has 1000 messages in inbox, we don't fetch 1000 messages in there.

What you need is remove `needaction_counter` contribution of the channel being deleted from the inbox counter.

For starred counter, there's currently no way to diff from deletion of message, as we don't track the amount of starred messages in a channel.

Reply:
Indeed, currently there is no proper solution to find out the starred messages counter for the deleted channel. So, the starred messages counter won't be updated. But it is quite an issue. So, I think we should find a way to resolve this in master. 

And Thanks for the suggestions!

Reply:
In master we should define `starredCounter` on thread, so that when removing the thread, we know its contribution on starred counter and diff accordingly